### DISCUSSION TITLE
How to download & use whisper's large-v2 model?

### DISCUSSION THREAD

**Discussion by Kirk2ko** [SubtitleEdit/subtitleedit#6581] [General]
I saw the large-v2 in action in another tool (youtube-whisperer) and it works really well for non-English languages.
https://huggingface.co/openai/whisper-large-v2/
I would like to use it Subtitle Edit as well, but I see that only the "original" large model is available for downloading... 
or the "large" now is actually the "large-v2" model? Thanks!

**Comment by niksedk** [comment]
In the current beta the `large model` for both `Whisper CPP` and ` original Whisper Python` points to the new large model (you might need to re-download)

Current beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.10/SubtitleEditBeta.zip

**Reply by Kirk2ko** [reply]
Awesome, will look into that. Thank you :)

**Reply by Kirk2ko** [reply]
Hi Nikolaj, Seems like I keep getting "no text found".
Any idea how to sort this?

**Reply by Sircam19** [reply]
Nikolaj, you do such wonderful work. Thank you!! Yes, I have been having the same issues for a little while now... I'll try the current beta also and report back as to the same error that @Kirk2ko has noted.

**Reply by niksedk** [reply]
You can attach the `error_log.txt` file and the audio file if you want me to take a closer look... (can be attached here as .zip or you can link/e-mail)

**Reply by Kirk2ko** [reply]
> You can attach the `error_log.txt` file and the audio file if you want me to take a closer look... (can be attached here as .zip or you can link/e-mail)

Where can I find this file? I couldn't locate it in the subtitle edit folder.

**Reply by niksedk** [reply]
> > You can attach the `error_log.txt` file and the audio file if you want me to take a closer look... (can be attached here as .zip or you can link/e-mail)
> 
> Where can I find this file? I couldn't locate it in the subtitle edit folder.

Press Ctrl+Shift+Alt+D in the SE main window, to open the SE data folder.

**Reply by Kirk2ko** [reply]
[error.zip](https://github.com/SubtitleEdit/subtitleedit/files/10373435/error.zip)
Thanks!
zip file attached with the latest error log + an audio file in Hebrew.
Do notice that when using the "large" whisper model - I get the "no text found" in every format (video / audio), also I've tested with both English and Hebrew files. None worked.
with the "base" model, however, I did manage to get the text.
But it's obviously less accurate for English and is no good for non-English languages, so I really need that large model to work...

Thank you again!

**Reply by darnn** [reply]
This is neither here nor there, but I recommend this, which has been linked by someone else here, in the meanwhile:
https://colab.research.google.com/github/ANonEntity/WhisperWithVAD/blob/main/WhisperWithVAD.ipynb

Does fairly okay with Hebrew. I wanted to try youtube-whisperer to see how it compares but it just timed out.

**Reply by niksedk** [reply]
> [error.zip](https://github.com/SubtitleEdit/subtitleedit/files/10373435/error.zip) Thanks! zip file attached with the latest error log + an audio file in Hebrew. Do notice that when using the "large" whisper model - I get the "no text found" in every format (video / audio), also I've tested with both English and Hebrew files. None worked. with the "base" model, however, I did manage to get the text. But it's obviously less accurate for English and is no good for non-English languages, so I really need that large model to work...
> 
> Thank you again!

OK, it looks like your model is only partial downloaded... you need to download the model again.
Latest beta now uses a temp file name while downloading + does more error checking: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.10/SubtitleEditBeta.zip

You can also download models manually via links found here: https://www.nikse.dk/subtitleedit/help#audio_to_text

I got this with the large model:
````
1
00:00:01,375 --> 00:00:04,000
היי, עניב כאן, ובואו נדבר על מה שבאמת חשוב

2
00:00:04,001 --> 00:00:05,000
בעסקים.

3
00:00:05,001 --> 00:00:08,000
אם אתם מנהלים עסק או חברה,
ואתם רוצים להגביר את המחירות,

4
00:00:09,000 --> 00:00:12,000
ליצור מהלכים שיווקים שבאמת מזיזים את הקהל,

5
00:00:12,475 --> 00:00:15,000
וגם לחסוך לעצמכם זמן עבודה יקר,

6
00:00:15,300 --> 00:00:17,000
אז אני אשמח להכיר אתכם ואת העסק שלכם יותר

7
00:00:17,001 --> 00:00:22,000
לעומק, ולתפור לכם באופן
אישי תוכנית פעולה לצמיחה

8
00:00:22,001 --> 00:00:27,000
שכוללת סרטון, או סדרת סרטונים
אסטרטגיים שיתיסו את העסק

9
00:00:27,001 --> 00:00:28,001
שלכם קדימה.

10
00:00:28,850 --> 00:00:32,000
אז אם הגעתם לנקודה שבה
מיציתם את השלב הנוכחי בעסק,

11
00:00:33,025 --> 00:00:35,000
ועכשיו אתם מחפשים לגדול,

12
00:00:35,200 --> 00:00:40,000
תלחצו על הכפתור כאן למטה כדי שנתאם
לנו שיחה, ואני כבר אספר לכם עוד כשנדבר.

13
00:00:40,001 --> 00:00:45,000
אתם עדיין פה? טוב, יש לי כמה שניות.

14
00:00:45,001 --> 00:00:47,000
*שירה.

````

**Reply by Kirk2ko** [reply]
Works now! thank you.
Seems like the problem stems from downloading the large model directly from Subtitle edit. It downloads, to my best guess, an incomplete or corrupted file.
That happened when I first tested Subtitle edit with a previous version around 2 months ago (I think it was 3.6.7) with the large model, and now with 3.6.10 after I re-downloaded the model from the software to get the latest version.

When downloaded the model manually, there was about 1GB gap (direct download: ~2GB file, manual: 3GB file) and now the model worked.

Anyway if anybody come across "no text found" with the large model, downloading manually would be the best bet.
